docs/backup/02-01-new-backup-strategy.md

```diff
@@ -0,0 +1,183 @@
+
+VolumeSnapshot support is only available for [CSI drivers](https://kubernetes-csi.github.io/docs/). However, not all the CSI drivers support the volume snapshot functionality. You can find a list of all the drivers with the supported functionalities [here](https://kubernetes-csi.github.io/docs/drivers.html).
+
+As an example, assume that you have a PVC `pvc-to-backup` which was created using a CSI-enabled StorageClass. Now you want to take a snapshot. You can trigger a snapshot by creating a VolumeSnapshot object like the following:
+
+> **NOTE:** The PVC to be backed up must be created using CSI-enabled Storage Class.
+
+```yaml
+apiVersion: snapshot.storage.k8s.io/v1beta1
+kind: VolumeSnapshot
+metadata:
+  name: volume-snapshot
+spec:
+  volumeSnapshotClassName: csi-snapshot-class
+  source:
+    persistentVolumeClaimName: pvc-to-backup
+```
+
+Now assume your PVC is corrupt, and you want to re-create it using the snapshot. Then, you can create it by using the snapshot you created before as the data source for the new PVC:
+
+```yaml
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: pvc-restored
+spec:
+  accessModes:
+    - ReadWriteOnce
+  storageClassName: csi-storage-class
+  resources:
+    requests:
+      storage: 10Gi
+  dataSource:
+    name: volume-snapshot
+    kind: VolumeSnapshot
+    apiGroup: snapshot.storage.k8s.io
+```
+
+This will create a new PVC `pvc-restored` with pre-populated data from the snapshot.
+
+You can find more information about VolumeSnapshots [here](https://kubernetes.io/docs/concepts/storage/volume-snapshots/).
+
+Currently none of the cloud providers support this API out-of-the-box yet. You must follow the instructions below to enable this feature for various providers.
+
+### AKS
+
+Minimum Kubernetes version supported is 1.17.
+
+Install CSI driver following [this documentation](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/master/docs/install-csi-driver-master.md).
+
+Then, you can follow [this example](https://github.com/kubernetes-sigs/azuredisk-csi-driver/tree/master/deploy/example/snapshot) to see how you can create a VolumeSnapshot.
+
+### GKE
+
+Minimum Kubernetes version supported is 1.14.
+
+Enable the required feature gate on the cluster following [this document](https://cloud.google.com/kubernetes-engine/docs/how-to/gce-pd-csi-driver#enabling_on_a_new_cluster).
+
+Install CSI driver following the instructions [here](https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blob/master/docs/kubernetes/user-guides/snapshots.md#kubernetes-snapshots-user-guide-alpha).
+
+Then, you can follow [this example](https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blob/master/docs/kubernetes/user-guides/snapshots.md#snapshot-example) to see how you can create a VolumeSnapshot.
+
+### Gardener Azure
```

Comment:
Let's have a general Gardener section explaining how to do it on Gardener (as for gcp and aws it is already available). Then have a note for Azure which we just need to remove later